src/Database/LSMTree/Internal/Merge.hs

```diff
@@ -199,13 +179,16 @@ close fs Merge {..} = do
 -- run they came from. This is important for resolving multiple entries with the
 -- same key into one.
 --
--- Construct with 'newReaders', then keep calling 'getFromReaders'.
--- If aborting early, remember to call 'closeReaders'!
+-- Construct with 'readersNew', then keep calling 'readersPop'.
+-- If aborting early, remember to call 'readersClose'!
 data Readers fhandle = Readers {
```

Comment:
So do you think a separate module would be better?

Reply:
> and how did you like `quickcheck-lockstep`?

There was quite a bit of ceremony. I'm sure all these parts have reasons for being there, to deal with issues arising in more complex examples, but for what I was doing (no variables, no opaque handle-like values that can't be compared etc.) it felt overkill. E.g. boilerplate like this:

```haskell
  observeReal _proxy = \case
      New {}          -> OEither . bimap OId OId
      PeekKey {}      -> OEither . bimap OId OId
      Pop {}          -> OEither . bimap OId (OTuple3 . trimap OId OId OId)
      DropWhileKey {} -> OEither . bimap OId (OTuple2 . bimap OId OId)
```

This is basically all `OId`, but if I just do that, the tests fail because it doesn't match `observeModel` (where I have to do all the bimap stuff for it to typecheck).

I could probably improve a few things, though. Let me push the tests to a separate branch, so you can have a look.

Reply:
https://github.com/IntersectMBO/lsm-tree/commit/b1e878ec7d0b190c317d637147dfa3b2fc1c32d9